.github/workflows/nightlies.yml

```diff
@@ -0,0 +1,310 @@
+          path: 'druntime'
+
+      - name: Clone phobos
+        uses: actions/checkout@v2
+        with:
+          repository: 'dlang/phobos'
+          path: 'phobos'
+
+      - name: Clone dlang.org
+        uses: actions/checkout@v2
+        with:
+          repository: 'dlang/dlang.org'
+          path: 'dlang.org'
+
+      # Fetch host compiler
+      - uses: dlang-community/setup-dlang@v1
+        name: Install host DMD
+
+      # Actually build the docs
+      - name: Build docs and man pages
+        shell: bash
+        run: |
+          set -euox pipefail
+
+          # Build minimal host compiler (sometimes not triggered by dlang.org/posix.mak)
+          make -f posix.mak -j2 -C druntime
+
+          # Build docs and include the man pages
+          make -f posix.mak -j2 -C dlang.org release
+          cp -r dmd/generated/docs/man dlang.org/web/
+
+      # Save the generated documentation for the target-specific builds
+      - name: Upload generated docs as a temporary artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: dmd-nightly-docs
+          path: dlang.org/web
+          retention-days: 1
+          if-no-files-found: error
+
+    runs-on: ubuntu-latest
+
+  # Build and package a new release for each platform
+  build-all-releases:
+    name: Build nightly for ${{ matrix.target }}  on ${{ matrix.os }}
+    needs: build-docs
+    timeout-minutes: 60
+
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-20.04
+            target: linux
+          - os: macos-10.15
+            target: osx
+          - os: windows-2019
+            target: windows
+          # FreeBSD is built on an additional VM
+          - os: macos-10.15
+            target: freebsd
+
+    steps:
+
+      #################################################################
+      # Install the system dependencies required to build and run
+      # the actual release scripts
+      #
+      # Linux implementation based on `linux_both` in build_all.d and
+      # some additional experimentation to get curl working
+      #
+      - name: Install dependencies for linux
+        if: matrix.target == 'linux'
+        shell: bash
+        run: |
+          set -euox pipefail
+
+          # Install base dependencies (including multlib support)
+          sudo dpkg --add-architecture i386
+          sudo apt -y update
+          sudo apt -y install --no-install-recommends \
+              build-essential \
+              ca-certificates \
+              curl \
+              dpkg-dev \
+              fakeroot \
+              g++ \
+              g++-multilib \
+              gcc \
+              git \
+              gpg \
+              gpg-agent \
+              libcurl4 \
+              libcurl4-openssl-dev \
+              libcurl4:i386 \
+              libxml2 \
+              make \
+              p7zip-full \
+              rpm \
+              rsync \
+              unzip \
+              xz-utils
```

Comment:
Some of those should already be included in the base image, but I've kept them just to be sure.